internal/controller/platform/dbMigratorJob.go

```diff
@@ -0,0 +1,222 @@
+	quarkusDatasourceJobsserviceUsername string
+	quarkusDatasourceJobsservicePassword string
+	quarkusFlywayJobsserviceSchemas      string
+}
+
+func getDBSchemaName(persistencePostgreSQL *operatorapi.PersistencePostgreSQL, defaultSchemaName string) string {
+	jdbcURL := persistencePostgreSQL.JdbcUrl
+
+	if len(jdbcURL) == 0 {
+		if persistencePostgreSQL.ServiceRef != nil && len(persistencePostgreSQL.ServiceRef.DatabaseSchema) > 0 {
+			return persistencePostgreSQL.ServiceRef.DatabaseSchema
+		}
+	} else {
+		_, a, found := strings.Cut(jdbcURL, "currentSchema=")
+
+		if found {
+			if strings.Contains(a, "&") {
+				b, _, found := strings.Cut(a, "&")
+				if found {
+					return b
+				}
+			} else {
+				return a
+			}
+		}
+	}
+	return defaultSchemaName
+}
+
+func NewDBMigratorJobData(ctx context.Context, client client.Client, platform *operatorapi.SonataFlowPlatform, pshDI services.PlatformServiceHandler, pshJS services.PlatformServiceHandler) (*DBMigratorJob, error) {
+	quarkusDatasourceDataindexJdbcUrl := ""
+	quarkusDatasourceDataindexUsername := ""
+	quarkusDatasourceDataindexPassword := ""
+	quarkusFlywayDataindexSchemas := ""
+	quarkusDatasourceJobsserviceJdbcUrl := ""
+	quarkusDatasourceJobsserviceUsername := ""
+	quarkusDatasourceJobsservicePassword := ""
+	quarkusFlywayJobsserviceSchemas := ""
+
+	migrateDbDataindex := pshDI.IsJobsBasedDBMigration()
+	if migrateDbDataindex {
+		quarkusDatasourceDataindexJdbcUrl = platform.Spec.Services.DataIndex.Persistence.PostgreSQL.JdbcUrl
+		quarkusDatasourceDataindexUsername, _ = services.GetSecretKeyValueString(ctx, client, platform.Spec.Services.DataIndex.Persistence.PostgreSQL.SecretRef.Name, platform.Spec.Services.DataIndex.Persistence.PostgreSQL.SecretRef.UserKey, platform)
+		quarkusDatasourceDataindexPassword, _ = services.GetSecretKeyValueString(ctx, client, platform.Spec.Services.DataIndex.Persistence.PostgreSQL.SecretRef.Name, platform.Spec.Services.DataIndex.Persistence.PostgreSQL.SecretRef.PasswordKey, platform)
+		quarkusFlywayDataindexSchemas = getDBSchemaName(platform.Spec.Services.DataIndex.Persistence.PostgreSQL, "defaultDi")
+	}
+	migrateDbJobsservice := pshJS.IsJobsBasedDBMigration()
+	if migrateDbJobsservice {
+		quarkusDatasourceJobsserviceJdbcUrl = platform.Spec.Services.JobService.Persistence.PostgreSQL.JdbcUrl
+		quarkusDatasourceJobsserviceUsername, _ = services.GetSecretKeyValueString(ctx, client, platform.Spec.Services.JobService.Persistence.PostgreSQL.SecretRef.Name, platform.Spec.Services.JobService.Persistence.PostgreSQL.SecretRef.UserKey, platform)
+		quarkusDatasourceJobsservicePassword, _ = services.GetSecretKeyValueString(ctx, client, platform.Spec.Services.JobService.Persistence.PostgreSQL.SecretRef.Name, platform.Spec.Services.JobService.Persistence.PostgreSQL.SecretRef.PasswordKey, platform)
+		quarkusFlywayJobsserviceSchemas = getDBSchemaName(platform.Spec.Services.JobService.Persistence.PostgreSQL, "defaultJs")
+	}
```

Comment:
Instead, please create one single function and call it from here instead of adding these big conditional blocks. They are almost the same thing.